File: gaming/src/main/java/com/oruga/gaming/dto/PublicacionDto.java
package com.oruga.gaming.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PublicacionDto {
    private Long id;
    private String titulo;
    private String contenido;
    private Long autorId;
    private String autorNombre;
    private Integer likes;
    private Integer cantidadComentarios;
    private String fechaCreacion;
    private String tiempoRelativo;
    private Boolean esAnuncio;
}

File: MyApplication3/app/src/main/java/com/example/myapplication/data/remote/dto/PublicacionDTO.kt
package com.example.myapplication.data.remote.dto

data class CrearPublicacionRequest(
    val titulo: String,
    val contenido: String,
    val imagenesUrls: String = "",
    val esAnuncio: Boolean = false
)

data class PublicacionResponse(
    val id: Long,
    val autorNombre: String,
    val titulo: String,
    val contenido: String,
    val imagenUrl: String,
    val imagenesUrls: String,
    val fechaPublicacion: String,
    val cantidadLikes: Int,
    val cantidadComentarios: Int,
    val esAnuncio: Boolean
)